validations.go

```diff
@@ -112,11 +119,16 @@ func toCodeHostingURL(file string, baseURL *url.URL) url.URL {
 	u := *baseURL
 	u.Path = path.Join(u.Path, file)
 
-	return u
+	// Check if the base URL itself it's a Git repository
+	if allowGitClone && isGitURL(&u) {
+		return u, true
+	}
+
+	return u, false
 }
 
-// fileExists returns true if the file resource exists.
-func (p *Parser) fileExists(u url.URL, network bool) (bool, error) {
+// Returns true if the file resource exists.
+func (p *Parser) fileExists(u url.URL, network bool, isGitRepo bool) (bool, error) {
```

Comment:
Not a fan of the signature of this function, but I suppose that with the `--external-checks` refactor we'd consolidate the two bools into a enum.

Something like `fileExists(url url.URL, checkMode CheckMode).